app/src/test/java/org/greenstand/android/TreeTracker/orgpicker/OrgPickerViewModelTest.kt

```diff
@@ -0,0 +1,51 @@
+package org.greenstand.android.TreeTracker.orgpicker
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import io.mockk.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.models.organization.OrgRepo
+import org.greenstand.android.TreeTracker.utils.FakeFileGenerator
+import org.greenstand.android.TreeTracker.utils.getOrAwaitValueTest
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class OrgPickerViewModelTest {
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val orgRepo = mockk<OrgRepo>(relaxed = true)
+    private lateinit var orgPickerViewModel: OrgPickerViewModel
+
+    @Before
+    fun setup(){
+        orgPickerViewModel = OrgPickerViewModel(orgRepo)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun `set fake organization, returns success with correct data`()= runBlockingTest {

```

Comment:
We should break things down into two test methods.
1. Test that the state instantly gets the latests org from the repo. This tests the logic in the `init` block.
2. Test that when we set a new org, the selected org in the state updates and the repo assigns that new org to itself. This tests the logic in the setOrg method.

Reply:
made adjustments.